decArgo_api/decoder_bindings/zip_nc_files.py

```diff
@@ -0,0 +1,40 @@
+"""Code to Zip decoded NC files."""
+
+import logging
+import os
+import traceback
+from io import BytesIO
+from pathlib import Path
+from zipfile import ZIP_DEFLATED, ZipFile
+
+
+class ZipNCFilesError(Exception):
+    """Raised when there is an issue zipping the NC files."""
+
+
+class ZipNCFiles:
+    """Methods to zip newly decoded NC files from the Coriolis Decoder."""
+
+    def __init__(self, wmonum: str):
+        """Initialise the instance with the wmonum and NC file location."""
+        self.wmonum = wmonum
+        self.base_output_location = Path(os.getenv("OUTPUT_LOCATION", f"/mnt/data/output/nc/{wmonum}")).resolve()
+
+    def zip_all_nc_files(self) -> BytesIO:
+        """Zip all .nc files under the wmonum's output directory.
+
+        Returns:
+            A BytesIO object to make available for download.
+        """
+        try:
+            buffer = BytesIO()
+
+            with ZipFile(buffer, "w", compression=ZIP_DEFLATED) as zipfile:
```

Comment:
ZIP_STORED might be preferable here, rather than spending extra CPU cycles on compressing the data before we send it. Not sure, though.